### ISSUE TITLE
sp.pl.masked_dotplot shows fewer genes than ProbesetSelector.plot_clf_genes

### ISSUE THREAD

**Issue by LouisK92** [theislab/spapros#230]
Some "spapros markers" are missing in `sp.pl.masked_dotplot` compared to `ProbesetSelector.plot_clf_genes`

```
import spapros as sp
adata = sp.ut.get_processed_pbmc_data()
selector = sp.se.ProbesetSelector(adata, "celltype", n=30, verbosity=0)
selector.select_probeset()

sp.pl.masked_dotplot(adata,selector)
selector.plot_clf_genes(celltypes=["CD4 T cells"],n_cols=4)
```

maybe secondary trees are not captured?

When fixed:

- [ ]  Add `selector.plot_clf_genes(celltypes=["CD4 T cells"],n_cols=4)` in basic selection tutorial

**Comment by LouisK92** [comment]
Also, maybe related, `sp.pl.masked_dotplot` misses some genes that are selected and are in `adata`. 

Steps to reproduce the behavior:
```python
marker_list = {
    'CD4 T cells': ['IL7R'], 
    'CD14+ Monocytes': ['CD14', 'LYZ'], 
    'B cells': ['MS4A1'], 
    'CD8 T cells': ['CD8A'], 
    'NK cells': ['GNLY', 'NKG7'], 
    'FCGR3A+ Monocytes': ['FCGR3A', 'MS4A7'], 
    'Dendritic cells': ['FCER1A', 'CST3'], 
    'Megakaryocytes': ['NAPA-AS1', 'PPBP'],
    'cell type A': ["Gene X", "Gene Y", "CAMK2N1"],
}

# create an instance of the ProbesetSelector class
selector = sp.se.ProbesetSelector(
    adata,
    n=None,
    celltype_key="celltype",
    verbosity=1,
    save_dir=None,
    marker_list=marker_list,
    n_list_markers={'adata_celltypes': 2, 'list_celltypes': 3},
)

# select probe set
selector.select_probeset()

sp.pl.masked_dotplot(adata,selector)
```

Missing genes after 'PPBP' like 'CD14', 'CD8A', 'IL7R'. 

When fixed:

- [ ]  Add `sp.pl.masked_dotplot(adata,selector)` in advanced selection tutorial use case iv.

**Comment by LouisK92** [comment]
Everything is fine for the `masked_dotplot`. 

The confusing part is that `selector.plot_clf_genes` doesn't use cell type specific importance scores and instead uses the maxima over cell types for each gene. I can't solve that quickly right now. Instead I added a printed note when plotting that describes the behavior, see commit 62f1548f49e2d04dd2a757eea2a28aa87207bcba.

Also note that `masked_dotplot` has default `imp_threshold = 0.05`. I.e. "wrong" importance scores are plotted in `plot_clf_genes` while `selector.genes_of_primary_trees` shows that `masked_dotplot` filtered properly based on cell type specific scores.

Closing for now, mentioned tutorials could be extended (would leave out the `plot_clf_genes` fct though), anyway, low priority for now.
